mobile/src/bundles/employer/employer-api.ts

```diff
@@ -0,0 +1,46 @@
+import { ApiPath, ContentType } from '~/bundles/common/enums/enums';
+import { type UserDetailsResponseDto } from '~/bundles/employer/types/types';
+import { UserDetailsApiPath } from '~/bundles/talent/enums/enums';
+import { HttpApiBase } from '~/framework/api/api';
+import { type Http } from '~/framework/http/http';
+import { type Storage } from '~/framework/storage/storage';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+    storage: Storage;
+};
+
+class EmployerApi extends HttpApiBase {
+    public constructor({ baseUrl, http, storage }: Constructor) {
+        super({ path: ApiPath.USER_DETAILS, baseUrl, http, storage });
+    }
+
+    public async getAllTalents(): Promise<UserDetailsResponseDto[]> {
+        const response = await this.load(
+            this.getFullEndpoint(UserDetailsApiPath.ROOT, {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+        return await response.json<UserDetailsResponseDto[]>();
+    }
+
+    public async getFilteredTalents(
+        payload: string,
+    ): Promise<UserDetailsResponseDto[]> {
+        const response = await this.load(
+            this.getFullEndpoint(UserDetailsApiPath.ROOT, payload, {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+        return await response.json<UserDetailsResponseDto[]>();
+    }
```

Comment:
I think it can be one request, filters will depend on query params

Reply:
Than we will always run create query string function before make request, thougt that it's unnesseray. 
Combine to one request?

Reply:
It's ok

Reply:
Let's combine

Reply:
It's a common thing to have for example default query params, so it should be the same